src/bin/pg_autoctl/fsm.c

```diff
@@ -362,6 +362,9 @@ KeeperFSMTransition KeeperFSM[] = {
 	 */
 	{ SECONDARY_STATE, REPORT_LSN_STATE, COMMENT_SECONDARY_TO_REPORT_LSN, &fsm_report_lsn },
 	{ CATCHINGUP_STATE, REPORT_LSN_STATE, COMMENT_SECONDARY_TO_REPORT_LSN, &fsm_report_lsn },
+	{ MAINTENANCE_STATE, REPORT_LSN_STATE, COMMENT_SECONDARY_TO_REPORT_LSN, &fsm_report_lsn },
+	{ PREPARE_MAINTENANCE_STATE, REPORT_LSN_STATE, COMMENT_SECONDARY_TO_REPORT_LSN, &fsm_report_lsn },
```

Comment:
We should allow going from PREPARE_MAINTENANCE_STATE to CATCHINGUP_STATE too.

Reply:
Good point. I'm not sure how to make it happen, but I can see the kind of “race condition” that could raise the situation. Fixed now.